namer/core/src/main/scala/io/buoyant/namer/RewritingNamerInitializer.scala

```diff
@@ -0,0 +1,24 @@
+package io.buoyant.namer
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.{Namer, Path}
+import com.twitter.finagle.Stack.Params
+import io.buoyant.namer.util.PathMatcher
+
+class RewritingNamerInitializer extends NamerInitializer {
+  override val configId = "io.l5d.rewrite"
+  val configClass = classOf[RewritingNamerConfig]
+}
+
+case class RewritingNamerConfig(
+  pattern: String,
+  name: String
+) extends NamerConfig {
+  assert(pattern != null)
+  assert(name != null)
+
```

Comment:
Even though we're not 100% on how we feel about this namer, it should be safe for production use from a performance point of view.  I would lean towards not marking it as experimental.